### DISCUSSION TITLE
Bug or feature: `opacity` vs `fillOpacity` for polygons?

### DISCUSSION THREAD

**Discussion by rdewit** [geostyler/geostyler#1510] [Q&A] [answered]
The SLD spec supports `fill-opacity` for polygons, Geostyler has `fillOpacity`, but the openlayers parser assumes the field is called `opacity` and doesn't know about `fillOpacity`, therefore cannot render transparent polygons converted from SLDs.

Is there consensus about what the internal name should be for the fill opacity of a polygon?

**Comment by jansule** [comment]
Thanks for opening the discussion @rdewit, that is definitely a point we should get straight.

In SLD (SE), there is a clear distinction between `fill-opacity` (used in `Fill` element), `stroke-opacity` (used in `Stroke` element), and `opacity`(used in `Graphic`, `RasterSymbolizer`, `ImageOutline`). When styling a polygon, one has to use the `fill-opacity` parameter and the `stroke-opacity` parameter, to style the inside and the border of the polygon, respectively. AFAIK, there is no general `opacity` element that sets the value for both fill and stroke.

So I think, we should follow the same approach in GeoStyler and get rid of the `opacity` property in `BaseSymbolizer` (https://github.com/geostyler/geostyler-style/blob/master/style.ts#L126). The `opacity` value is a result of the early stages of geostyler-style, where an overall opacity property was defined for all symbolizers. Later, we added the more precise properties.

This change would probably affect multiple repositories, including https://github.com/geostyler/geostyler, where we currently provide 3 input fields for specifying the opacity of a fillSymbolizer `Fill-Opacity`, `Opacity`, `Stroke-Opacity` - with `Opacity` apparently having no impact on the style.

It would also be great to get the opinions of at least @dnlkoch @KaiVolland @chrismayer.

Some related issues:
- https://github.com/geostyler/geostyler/issues/1359
- https://github.com/geostyler/geostyler/issues/915

**Reply by KaiVolland** [reply]
The `opacity` is more or less the overall opacity of the Symbolizer. It can be an additonal opacity when used in combination with `fill-opacity` and `stroke-opacity`. Even if it is not the way SLD handles it, i can imagine styles (e.g. OL) where it make sense to combine all those three values.

**Reply by jansule** [reply]
Do you see the overall opacity as just an UI tweak for more convenience, or as an actual value for the style definition?

How would you handle setting values in parsers, where there is no overall opacity e.g. SLD?

**Reply by KaiVolland** [reply]
We may just add it to the `unsupportedProperties`.

**Reply by rdewit** [reply, marked as answer]
I can confirm that the `fill-opacity` vs `opacity` issue within the openlayers parser has been fixed by https://github.com/geostyler/geostyler-openlayers-parser/pull/757. Thanks, @KaiVolland !
